### PR TITLE
Convert tags to lower case

### LINKED ISSUE #437: [FEATURE REQUEST] Tag search should not be case sensitive
When searching with tags the current implementation is search input tag should match exactly including case, but that should not be the case.

### PR DESCRIPTION
**Pull request by Mikearaya**
Closes #437 

- Added custom LocaleCaseString scalar that does the conversion on GraphQL API level
- Convert tags to lower case for all entities in bulk import 
- Added migration for all collections tag fields that convert them to lowercase.

while working on this, I noticed a minor issue with findFilters typing so I have fixed that too

### CODE REVIEW

**Review by pozylon — COMMENTED** [review]
In general:
- Looks good conceptually!
- User also has tags but no migration?
- I don't like the utils function "convertTagsToLowerCase" because it is a util that is only needed just 1 time during migrations and adding it to a package as util function somehow does not feel right.

File: packages/core-users/src/module/addMigrations.ts [hunk 1]
```diff
@@ -1,7 +1,27 @@
...
+    if (count % 500 === 0) {
+      bulk.execute();
+      bulk = collection.initializeUnorderedBulkOp();
+      count = 0;
+    }
+  }
+  if (count > 0) bulk.execute();
+};
+
 export default function addMigrations(repository: MigrationRepository<Migration>) {
```
Comment (Mikearaya) [inline packages/core-users/src/module/addMigrations.ts:25]: @pozylon User has a migration too

**Comment by pozylon** [comment]
@Mikearaya We need an admin ui branch to fix the breaking changes with the tags, can you prepare that too so we can merge both of that together when ready?

**Review by pozylon — APPROVED** [review]

**Comment by Mikearaya** [comment]
@pozylon prepared we can merge it when this is merged https://github.com/xecutors/unchained-adminui/pull/301

**Pull request merged**